proposed/http-message.md

```diff
@@ -1385,10 +1385,12 @@ interface UriInterface
      * This method MUST retain the state of the current instance, and return
      * an instance that contains the specified path.
      *
-     * The path MUST be prefixed with "/"; if not, the implementation MAY
-     * provide the prefix itself.
+     * The path can either be empty or absolute (starting with a slash) or
+     * rootless (not starting with a slash). Implementations MUST support all
+     * three syntaxes.
      *
-     * An empty path value is equivalent to removing the path.
+     * Users can provide the path both in encoded as well as decoded form.
```

Comment:
is this correct English? "both ... as well" sounds strange, I think I have to change it to "both in encoded and in decoded form".